lib/primitives/tx.js

```diff
@@ -950,13 +952,9 @@ class TX {
       const controlBlock = input.witness.getControlBlock();
       const leafVersion = controlBlock[0] & 0xfe;
 
-      const tapleaf = input.witness.getTapleaf();
-      const size = encoding.sizeVarBytes(tapleaf);
-      const serializedTapleaf = bio.write(size + 1);
-
-      serializedTapleaf.writeU8(leafVersion);
-      serializedTapleaf.writeVarBytes(tapleaf);
-      bw.writeHash(taggedHash.TapLeafHash.digest(serializedTapleaf.render()));
+      const script = new Script({raw: input.witness.getTapleaf()});
```

Comment:
Maybe the `TapLeaf` class can have a helper for this:

```js
static fromRawScript(data, leafVersion) {
  return new this(Script.fromRaw(data), leafVersion);
}
```

...this can be applied in `verifyTaprootCommitment()` as well I think

Reply:
Problem: Since we are using Taproot inside script.js, we cannot import anything from script to taproot, as it causes circular dependency...
Solution1: Either ignore this use case.
Solution2: Merge the taproot file into script.js, as we discussed earlier.

Reply:
Gotcha. I wonder if we should move verifyTaprootCommitment() to taproot.js ?

Reply:
Yeah... that could work,
We dont need to imprt taproot to script then...

Reply:
Yeah I think that's an OK move, its a `static` function anyway and doesn't use any other property of `Script` (I don't think...?) Also looks like the only code that could be moved from script.js to taproot.js

Reply:
The circular dependency will still remain --> https://github.com/bcoin-org/bcoin/blob/taproot/lib/script/script.js#L3416

Reply:
Ok you're right, for now this is as clean as that snippet can get ;-)

Let's maybe just use a helper function for clarity:
```js
      const script = Script.fromRaw(input.witness.getTapleaf());
```